src/docfx/build/link/LinkResolver.cs

```diff
@@ -218,8 +221,13 @@ internal class LinkResolver
 
             if (relativePath.StartsWith("~/") || relativePath.StartsWith("~\\"))
             {
-                // Treat ~/ as path relative to docset
-                pathToDocset = new PathString(relativePath.Substring(2).TrimStart('/', '\\'));
+                var (_, metadata) = _metadataProvider.GetMetadata(referencingFile);
+                pathToDocset = new PathString(Path.GetRelativePath(
```

Comment:
Consider only do the `Path.Combine` here and move line 238 - line 242 to line 244

Reply:
right